src/containers/Tenant/Diagnostics/TopShards/TopShards.tsx

```diff
@@ -66,126 +69,119 @@ function stringToQuerySortOrder(value) {
-                    path: currentSchemaPath,
-                    sortOrder: stringToQuerySortOrder(sortOrder),
-                }),
-            );
-        }
+export const TopShards = ({path, type}: TopShardsProps) => {
+    const dispatch = useDispatch();
 
-        return () => {
-            autofetcher.stop();
-        };
-    }, [autorefresh, currentSchemaPath, path, sendShardQuery, sortOrder]);
+    const {
+        autorefresh,
+        currentSchema: {
+            Path: currentSchemaPath = undefined
+        } = {}
+    } = useTypedSelector((state) => state.schema);
 
-    // don't show loader for requests triggered by table sort, only for path change
-    useEffect(() => {
-        setShardQueryOptions({
-            wasLoaded: false,
-            data: undefined,
-        });
-    }, [currentSchemaPath, path, setShardQueryOptions]);
+    const {
+        loading,
+        data: {result: data = undefined} = {},
+        error,
+        wasLoaded,
+    } = useTypedSelector((state) => state.shardsWorkload);
+
+    const [sortOrder, setSortOrder] = useState(tableColumnsNames.CPUCores);
+
+    useAutofetcher(() => {
+        dispatch(
+            sendShardQuery({
+                database: path,
+                path: currentSchemaPath,
+                sortOrder: stringToQuerySortOrder(sortOrder),
+            })
+        );
+    }, [dispatch, currentSchemaPath, path, sortOrder], autorefresh);
 
+    // don't show loader for requests triggered by table sort, only for path change
     useEffect(() => {
-        sendShardQuery({
-            database: path,
-            path: currentSchemaPath,
-            sortOrder: stringToQuerySortOrder(sortOrder),
-        });
-    }, [currentSchemaPath, path, sendShardQuery, sortOrder]);
+        dispatch(
+            setShardQueryOptions({
+                wasLoaded: false,
+                data: undefined,
+            })
+        );
+    }, [dispatch, currentSchemaPath, path]);
 
     const history = useContext(HistoryContext);
 
-    const onSchemaClick = (schemaPath) => {
-        return () => {
-            setCurrentSchemaPath(schemaPath);
-            getSchema({path: schemaPath});
-            history.go(0);
-        };
-    };
-
-    const onSort = (newSortOrder) => {
+    const onSort = (newSortOrder?: SortOrder | SortOrder[]) => {
         // omit information about sort order to disable ASC order, only DESC makes sense for top shards
         // use a string (and not the DataTable default format) to prevent reference change,
         // which would cause an excess state change, to avoid repeating requests
         setSortOrder(dataTableToStringSortOrder(newSortOrder));
     };
 
-    const tableColumns = useMemo(() => {
+    const tableColumns: Column<any>[] = useMemo(() => {
+        const onSchemaClick = (schemaPath: string) => {
+            return () => {
+                dispatch(setCurrentSchemaPath(schemaPath));
+                dispatch(getSchema({path: schemaPath}));
```

Comment:
Do we need to do a fetch here, if schema is fetched anyway on page reload?

Reply:
Most likely we do not. But it’s another matter, I’d prefer not to sort it out in a refactoring PR